Failed to Build Single Runtime

Run `cargo build --features evm-tracing` under the runtime folder.

```
error: cannot find macro `thread_local` in this scope
  --> primitives/evm-trace/events/src/lib.rs:49:1
   |
49 | environmental::environmental!(listener: dyn Listener + 'static);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `thread_local` is in scope, but it is an attribute: `#[thread_local]`
   = note: this error originates in the macro `$crate::thread_local_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find value `GLOBAL` in this scope
  --> primitives/evm-trace/events/src/lib.rs:49:1
   |
49 | environmental::environmental!(listener: dyn Listener + 'static);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
   = note: this error originates in the macro `environmental::environmental` (in Nightly builds, run with -Z macro-backtrace for more info)
```

--- a/node/runtime/pangolin/Cargo.toml
+++ b/node/runtime/pangolin/Cargo.toml
@@ -241,7 +241,10 @@ std = [
 	"sp-transaction-pool/std",
 	"sp-version/std",
 	# trace
+	"dp-evm-tracer/std",
 	"dp-evm-trace-apis/std",
+	"rlp/std",
+	"sha3/std",
 	# frontier
 	"fp-evm/std",
 	"fp-rpc/std",

--- a/node/runtime/pangoro/Cargo.toml
+++ b/node/runtime/pangoro/Cargo.toml
@@ -102,9 +102,9 @@ fp-evm                       = { default-features = false, git = "https://github
 fp-rpc                       = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "darwinia-v0.12.1-dev" }
 fp-self-contained            = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "darwinia-v0.12.1-dev" }
 fp-storage                   = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "darwinia-v0.12.1-dev" }
-pallet-evm-precompile-simple = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "darwinia-v0.12.1-dev" }
 pallet-base-fee              = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "darwinia-v0.12.1-dev" }
 pallet-evm                   = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "darwinia-v0.12.1-dev" }
+pallet-evm-precompile-simple = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "darwinia-v0.12.1-dev" }
 
 
 [build-dependencies]
@@ -196,6 +196,9 @@ std = [
 	"sp-version/std",
 	# trace
 	"dp-evm-trace-apis/std",
+	"dp-evm-tracer/std",
+	"rlp/std",
+	"sha3/std",
 	# frontier
 	"fp-evm/std",
 	"fp-rpc/std",

--- a/primitives/evm-trace/apis/Cargo.toml
+++ b/primitives/evm-trace/apis/Cargo.toml
@@ -11,6 +11,7 @@ version     = "2.8.5"
 
 [dependencies]
 # crates.io
+codec          = { package = "parity-scale-codec", version = "2.3", default-features = false }
 environmental  = { version = "1.1", default-features = false }
 ethereum       = { version = "0.11.1", default-features = false, features = ["with-codec"] }
 ethereum-types = { version = "0.12", default-features = false }
@@ -18,7 +19,6 @@ hex            = { version = "0.4", features = ["serde"], optional = true }
 serde          = { version = "1.0", features = ["derive"], optional = true }
 serde_json     = { version = "1.0", optional = true }
 # paritytech
-codec      = { package = "parity-scale-codec", version = "2.2", default-features = false }
 sp-api     = { default-features = false, git = "https://github.com/darwinia-network/substrate", branch = "darwinia-v0.12.1" }
 sp-core    = { default-features = false, git = "https://github.com/darwinia-network/substrate", branch = "darwinia-v0.12.1" }
 sp-io      = { default-features = false, git = "https://github.com/darwinia-network/substrate", branch = "darwinia-v0.12.1" }
@@ -30,6 +30,7 @@ sp-std     = { default-features = false, git = "https://github.com/darwinia-netw
 default = ["std"]
 std = [
 	# crates.io
+	"codec/std",
 	"ethereum/std",
 	"ethereum-types/std",
 	"environmental/std",
@@ -37,7 +38,6 @@ std = [
 	"serde",
 	"serde_json",
 	# paritytech
-	"codec/std",
 	"sp-api/std",
 	"sp-core/std",
 	"sp-io/std",

--- a/primitives/evm-trace/ext/Cargo.toml
+++ b/primitives/evm-trace/ext/Cargo.toml
@@ -11,11 +11,11 @@ version     = "2.8.5"
 
 [dependencies]
 # crates.io
+codec          = { package = "parity-scale-codec", version = "2.3", default-features = false }
 ethereum-types = { version = "0.12.0", default-features = false }
 # darwinia-network
 dp-evm-trace-events = { default-features = false, path = "../events" }
 # paritytech
-codec                = { package = "parity-scale-codec", version = "2.2", default-features = false }
 sp-externalities     = { default-features = false, git = "https://github.com/darwinia-network/substrate", branch = "darwinia-v0.12.1" }
 sp-runtime-interface = { default-features = false, git = "https://github.com/darwinia-network/substrate", branch = "darwinia-v0.12.1" }
 sp-std               = { default-features = false, git = "https://github.com/darwinia-network/substrate", branch = "darwinia-v0.12.1" }
@@ -24,11 +24,11 @@ sp-std               = { default-features = false, git = "https://github.com/dar
 default = ["std"]
 std = [
 	# crates.io
+	"codec/std",
 	"ethereum-types/std",
 	# darwinia-network
 	"dp-evm-trace-events/std",
 	# paritytech
-	"codec/std",
 	"sp-externalities/std",
 	"sp-runtime-interface/std",
 	"sp-std/std",

--- a/primitives/evm-trace/tracer/Cargo.toml
+++ b/primitives/evm-trace/tracer/Cargo.toml
@@ -10,6 +10,7 @@ version     = "2.8.5"
 
 [dependencies]
 # crates.io
+codec          = { package = "parity-scale-codec", version = "2.3", default-features = false }
 ethereum-types = { version = "0.12.0", default-features = false }
 evm            = { version = "0.33.1", default-features = false, features = ["with-codec"] }
 evm-gasometer  = { version = "0.33", default-features = false }
@@ -18,7 +19,6 @@ evm-runtime    = { version = "0.33", default-features = false }
 dp-evm-trace-events = { path = "../events", default-features = false, features = ["evm-tracing"] }
 dp-evm-trace-ext    = { path = "../ext", default-features = false }
 # paritytech
-codec      = { package = "parity-scale-codec", version = "2.2", default-features = false }
 sp-core    = { default-features = false, git = "https://github.com/darwinia-network/substrate", branch = "darwinia-v0.12.1" }
 sp-io      = { default-features = false, git = "https://github.com/darwinia-network/substrate", branch = "darwinia-v0.12.1" }
 sp-runtime = { default-features = false, git = "https://github.com/darwinia-network/substrate", branch = "darwinia-v0.12.1" }

